src/Formik.tsx

```diff
@@ -131,12 +131,20 @@ export class Formik<Values = FormikValues> extends React.Component<
     });
   };
 
-  setValues = (values: FormikState<Values>['values']) => {
-    this.setState({ values }, () => {
-      if (this.props.validateOnChange) {
-        this.runValidations(values);
-      }
-    });
+  setValues = (values: ValueUpdate<Values>) => {
+    if (isFunction(values)) {
+      this.setState({ values: values(this.state.values) }, () => {
+        if (this.props.validateOnChange) {
+          this.runValidations(values);
+        }
+      });
+    } else {
+      this.setState({ values }, () => {
+        if (this.props.validateOnChange) {
+          this.runValidations(values);
+        }
+      });
+    }
```

Comment:
Since both branched do the same, extract `nextValues`.

```js
const nextValues = isFunction(valuesOrCallback) ? valuesOrCallback(this.state.values) : valuesOrCallback;

this.setState({ values: nextValues }, () => {
	if (this.props.validateOnChange) {
		this.runValidations(nextValues);
	}
})
```

Reply:
That should be fixed now :)